trust-store-trust-policy-specification.md

```diff
@@ -146,37 +145,70 @@ Trust policy for the scenario where ACME Rockets uses artifacts signed by themse
   - **`registryScopes`**(*array of strings*): This REQUIRED property determines which trust policy is applicable for the given artifact.
     The scope field supports filtering based on fully qualified repository URI `${registry-name}/${namespace}/${repository-name}`.
     For more information, see [registry scopes constraints](#registry-scopes-constraints) section.
-  - **`signatureVerification`**(*string*): This REQUIRED property dictates how signature verification is performed. Supported values are `strict`, `permissive`, `audit` and `skip`. Detailed explaination of each value is present [here](#signatureverification-details). A custom level can be defined by referencing a supported level value, and overriding individual validation checks. NOTE: Custom signature verification level will not be supported in `notation` RC1.
-  - **`trustStore`**(*string*): This REQUIRED property specifies a named trust store. Uses the format `{trust-store-type}:{named-store}`. Currently supported values for `trust-store-type` are `ca` and `tsa`. **NOTE**: When support for publicly trusted TSA is available, `tsa:publicly-trusted-tsa` is the default value, and implied without explictly specifying it. If a custom TSA is used the format `ca:acme-rockets,tsa:acme-tsa` is supported to specify it.
+  - **`verificationLevel`**(*string*): This OPTIONAL property dictates how signature verification is performed. Either `verificationLevel` or `verification` property MUST be specified. Supported values are `strict`, `permissive`, `audit` and `skip`. Detailed explaination of each value is present [here](#signatureverification-details).
+  - **`verification`**(*object*): This OPTIONAL property dictates how signature verification is performed using a [custom verification level](#custom-verification-level). Either `verificationLevel` or `verification` property MUST be specified.
```

Comment:
What if `verificationLevel` and `verification` both exist?

Reply:
Multi-typed property is hard to be implemented correctly. Can we avoid multi-typed properties?

Reply:
Agree with the additional complexity, but other options make the user experience less desirable. You either require two separate properties or set value to an object even for simple configurations. We are using the multi-typed property in this case as we see a benefit in user experience at the cost of implementation complexity.

Alternate 1 - Use one of the following to set a level, or to override a level. Uses two different top level properties.
```
     "signatureVerificationLevel" : "strict"

      OR

    "signatureVerification": {
      "level" : "strict",
      "override" : {
        "expiry" : "log"
      }
    }
```

Alternate 2 - use the same object definition, but makes the simple configuration which most users will use, more complicated.
```
"signatureVerification": {
     "level" : "strict"
}

OR

  "signatureVerification": {
    "level" : "strict",
    "override" : { 
      "expiry" : "log"
    }
  }

```

Reply:
My recommendation would be for uniform objects even for usability it would avoid issues when one person changes it from audit to strict 

```json
"signatureVerification" : {
      "level" : "audit" 
}
```

if we change it to strict we won't get mutually exclusive properties.

Reply:
Updated `signatureVerification` to an object type and not a multi-typed attribute.